tests/integration/components/bs-button-test.js

```diff
@@ -199,6 +199,65 @@ module('Integration | Component | bs-button', function(hooks) {
     assert.dom('button').hasText('default text');
   });
 
+  test('button is disabled while in pending state', async function(assert) {
+    let deferredClickAction = defer();
+    this.set('clickAction', () => {
+      return deferredClickAction.promise;
+    });
+
+    await render(hbs`{{bs-button onClick=clickAction}}`);
+    assert.dom('button').isNotDisabled();
+
+    await click('button');
+    assert.dom('button').isDisabled();
+
+    deferredClickAction.resolve();
+    await settled();
+    assert.dom('button').isNotDisabled();
+  });
+
+  test('button is not disabled while in pending state if preventConcurrency is false', async function(assert) {
+    let deferredClickAction = defer();
+    this.set('clickAction', () => {
+      return deferredClickAction.promise;
+    });
+
+    await render(hbs`<BsButton @onClick={{clickAction}} @preventConcurrency={{false}} />`);
+    await click('button');
+    assert.dom('button').isNotDisabled();
+
+    deferredClickAction.resolve();
+    await settled();
+  });
+
+  test('setting @disabled property to false prevents button from being disabled while in pending state', async function(assert) {
+    let deferredClickAction = defer();
+    this.set('clickAction', () => {
+      return deferredClickAction.promise;
+    });
+
+    await render(hbs`<BsButton @disabled={{false}} @onClick={{clickAction}} />`);
+    await click('button');
+    assert.dom('button').isNotDisabled();
+
+    deferredClickAction.resolve();
+    await settled();
+  });
+
+  test('setting disabled HTML attribute to false prevents button from being disabled while in pending state', async function(assert) {
+    let deferredClickAction = defer();
+    this.set('clickAction', () => {
+      return deferredClickAction.promise;
+    });
+
+    await render(hbs`<BsButton @onClick={{clickAction}} disabled={{false}} />`);
+    await click('button');
+    assert.dom('button').isNotDisabled();
+
+    deferredClickAction.resolve();
+    await settled();
+  });
```

Comment:
This test case is mainly about preventing a regression when we refactor to outer HTML syntax and spread attributes. Basically these test should ensure that we are applying default value for `disabled` before `...attributes` and not after.

Reply:
Nice!

But this test is actually failing for Ember 2.18, pretty sure due to a bug in the polyfill: https://github.com/rwjblue/ember-angle-bracket-invocation-polyfill/issues/69

Because of this I skipped some tests for Ember <3.4 (i.e. versions w/o native angle bracket support) here: https://github.com/kaliber5/ember-bootstrap/commit/a46ed2e1ea62fd4c451655c14aea6ec0a8520956#diff-fd815e23c0edac5d5ea76999058689dd. We should do the same here.